cmd/maya-apiserver/app/server/snapshot_endpoint.go

```diff
@@ -0,0 +1,180 @@
+package server
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/openebs/maya/types/v1"
+	"github.com/openebs/maya/volume/provisioners/jiva"
+)
+
+// SnapshotSpecificGetRequest deals with HTTP GET request w.r.t a Volume Snapshot
+func (s *HTTPServer) SnapshotSpecificRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	// Extract info from path after trimming
+	path := strings.TrimPrefix(req.URL.Path, "/latest/snapshot")
+
+	// Is req valid ?
+	if path == req.URL.Path {
+		fmt.Println("Request coming", path)
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+
+	switch {
+	case strings.Contains(path, "/create/"):
+		return s.SnapshotCreate(resp, req)
+	case strings.Contains(path, "/revert/"):
+		return s.SnapshotRevert(resp, req)
+	case strings.Contains(path, "/list"):
+		volName := strings.TrimPrefix(path, "/list/")
+		return s.SnapshotList(resp, req, volName)
+	default:
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+}
+
+// SnapshotCreate is http handler which handles snaphsot-create request
+func (s *HTTPServer) SnapshotCreate(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+
+	if req.Method != "PUT" && req.Method != "POST" {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+
+	fmt.Println("[DEBUG] Processing Volume Snapshot request")
+
+	snap := v1.VolumeSnapshot{}
+
+	// The yaml/json spec is decoded to pvc struct
+	if err := decodeBody(req, &snap); err != nil {
+
+		return nil, CodedError(400, err.Error())
+	}
+
+	// SnapshotName is expected to be available even in the minimalist specs
+	if snap.Metadata.Name == "" {
+		return nil, CodedError(400, fmt.Sprintf("Snapshot name missing in '%v'", snap.SnapshotName))
+	}
+
+	// Name is expected to be available even in the minimalist specs
+	if snap.Spec.VolumeName == "" {
+
+		return nil, CodedError(400, fmt.Sprintf("PVC Volume name missing in '%v'", snap.Spec.VolumeName))
+	}
+
+	fmt.Println("Volume Name :", snap.Spec.VolumeName)
+	fmt.Println("[DEBUG] Processing snapshot-create request of volume:", snap.Spec.VolumeName)
+
+	voldetails, err := s.vsmRead(resp, req, snap.Spec.VolumeName)
+	if err != nil {
+		return nil, err
+	}
+
+	ControllerIP := voldetails.Annotations["vsm.openebs.io/controller-ips"]
```

Comment:
Can you raise an issue to extract controller IPs, replica IPs from volume struct ?

Reply:
done